plancontext/platform.go

```diff
@@ -5,28 +5,29 @@ import (
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-var (
-	// Default platform.
-	// FIXME: This should be auto detected using buildkit
-	defaultPlatform = specs.Platform{
-		OS:           "linux",
-		Architecture: "amd64",
-	}
-)
+// DefaultPlatform to fallback in case it can't be automatically detected.
+var DefaultPlatform = specs.Platform{
+	OS:           "linux",
+	Architecture: "amd64",
+}
 
 type platformContext struct {
-	platform specs.Platform
+	platform *specs.Platform
 }
 
-func (c *platformContext) Get() specs.Platform {
+func (c *platformContext) Get() *specs.Platform {
```

Comment:
Is it not more accurate if you directly return the value there? We do not want to update the pointer outside `SetString` or `Set`

Reply:
I had to change this into a pointer due to this check (https://github.com/dagger/dagger/pull/2028/files#diff-bd3a55a72186f59e2e63efb4951573b2f9e4a7cc98086e922b0859f8ccc1dd09R86)  since we could have three potential states now: 

1 - Platform undefined
2 - Platform defined in Plan file
3 - Platform automatically detected. 

The only way to know if the platform should be detected automatically because it wasn't specifically set in the plan file is  to make that Platform variable a pointer. Thoughts?

Reply:
I understand, you couldn't just have a `isSet` boolean to avoid using a pointer?